lib/iris/tests/unit/analysis/test_Aggregator.py

```diff
@@ -315,6 +319,39 @@ def test_kwarg_pass_through_combined_kwargs(self):
         expected_kwargs.update(call_kwargs)
         lazy_func.assert_called_once_with(data, axis, **expected_kwargs)
 
+class Test_aggregated_by(tests.IrisTest):
+    def setUp(self):
+        # This sets up a cube with a time coordinate containing two years and
+        # two months.
+        latitude = DimCoord(np.linspace(-90, 90, 4),
+                            standard_name='latitude',
+                            units='degrees')
+        longitude = DimCoord(np.linspace(45, 360, 8),
+                             standard_name='longitude',
+                             units='degrees')
+        time_unit = unit.Unit('days since 1970-01-01 00:00:00',
+                              calendar=unit.CALENDAR_360_DAY)
+        time = DimCoord(np.linspace(0, 750, 26),
+                        standard_name='time',
+                        units=time_unit)
+        self.time_cube = Cube(np.zeros((26, 4, 8), np.float32),
+                               dim_coords_and_dims=[(time, 0),
+                                                    (latitude, 1),
+                                                    (longitude, 2)])
+
+    def test_month_name(self):
+        # This will test the ability of the aggregator to correctly calculate
+        # the order of months in the new coordinate array.
+        # The month names should appear in chronological order from January
+        # through to December.
+        coord_categorisation.add_month_number(self.time_cube, 'time',
+                                              name='month')
+        agg_cube = self.time_cube.aggregated_by('month', MEAN)
+        coord_categorisation.add_month(agg_cube, 'time', name='month_name')
+        self.assertEquals(agg_cube.coord('month_name').points[0], 'Jan')
```

Comment:
Well it is, yes.  The bit which decides how to group the months over which to aggregate depends on a mean calculation of the months.  I don't really understand why it is done like this or how this normally calculates the correct grouping as it is, but when there are an uneven number of months in the calculation it most definitely produces the wrong answer.

I'll try and explain that a bit better; the mean calculation takes each coord point for January and takes the average, then does the same for February, then March, etc.  In the case described here, there are three coord points for January, three for February, and only two for March and the rest of the months.  That means that when it takes the mean coord point from these groups, it looks like this:
[ 360.  390.  240.  270.  300.  330.  360.  390.  420.  450.  480.  510.]

The third point in this list is much lower than the first two, and is translated into a date coordinate as being in September of the previous year, which is clearly wrong, as are the rest of the data points determined here.

As I don't really understand how this whole section works anyway, I am really struggling with how to approach a fix for this.